source/state_representation/test/tests/space/cartesian/test_cartesian_pose.cpp

```diff
@@ -0,0 +1,277 @@
+  EXPECT_EQ(static_cast<CartesianState&>(random).get_wrench().norm(), 0);
+}
+
+TEST(CartesianPoseTest, CopyPose) {
+  CartesianPose pose1 = CartesianPose::Random("test");
+  CartesianPose pose2(pose1);
+  EXPECT_EQ(pose1.get_name(), pose2.get_name());
+  EXPECT_EQ(pose1.get_reference_frame(), pose2.get_reference_frame());
+  EXPECT_TRUE(pose1.data().isApprox(pose2.data()));
+  EXPECT_EQ(static_cast<CartesianState&>(pose2).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose2).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose2).get_wrench().norm(), 0);
+  CartesianPose pose3 = pose1;
+  EXPECT_EQ(pose1.get_name(), pose3.get_name());
+  EXPECT_EQ(pose1.get_reference_frame(), pose3.get_reference_frame());
+  EXPECT_TRUE(pose1.data().isApprox(pose3.data()));
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose3).get_wrench().norm(), 0);
+  // try to change non pose variables prior to the copy, those should be discarded
+  static_cast<CartesianState&>(pose1).set_twist(Eigen::VectorXd::Random(6));
+  static_cast<CartesianState&>(pose1).set_accelerations(Eigen::VectorXd::Random(6));
+  static_cast<CartesianState&>(pose1).set_wrench(Eigen::VectorXd::Random(6));
+  CartesianPose pose4 = pose1;
+  EXPECT_TRUE(pose1.data().isApprox(pose4.data()));
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose4).get_wrench().norm(), 0);
+  // copy a state, only the pose variables should be non 0
+  CartesianPose pose5 = CartesianState::Random("test");
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_twist().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_accelerations().norm(), 0);
+  EXPECT_EQ(static_cast<CartesianState&>(pose5).get_wrench().norm(), 0);
+
+  CartesianPose pose6;
+  EXPECT_TRUE(pose6.is_empty());
+  CartesianPose pose7 = pose6;
+  EXPECT_TRUE(pose7.is_empty());
+}
+
+TEST(CartesianPoseTest, SetData) {
+  CartesianPose cp1 = CartesianPose::Identity("test");
+  CartesianPose cp2 = CartesianPose::Random("test");
+  cp1.set_data(cp2.data());
+  EXPECT_TRUE(cp2.data().isApprox(cp1.data()));
+
+  auto pose_vec = cp2.to_std_vector();
+  cp1.set_data(pose_vec);
+  for (std::size_t j = 0; j < pose_vec.size(); ++j) {
+    EXPECT_FLOAT_EQ(pose_vec.at(j), cp1.data()(j));
+  }
+}
+
+TEST(CartesianPoseTest, CartesianPoseToStdVector) {
+  CartesianPose cp = CartesianPose::Random("test");
+  std::vector<double> vec_data = cp.to_std_vector();
+  EXPECT_EQ(vec_data.size(), 7);
+  for (size_t i = 0; i < vec_data.size(); ++i) {
+    EXPECT_EQ(cp.data()(i), vec_data[i]);
+  }
+}
+
+TEST(CartesianPoseTest, MultiplyTransformsBothOperators) {
```

Comment:
I think a lot of these `tf1` x `tf2` test cases could be simplified using a TEST_F fixture class. You can pre-instantiante your two poses with the right names and default position / orientation, then just call `set_pose` in one line. Would save a whole lot of setup for these 6-8 test cases